source/extensions/filters/http/jwt_authn/authenticator.cc

```diff
@@ -286,6 +288,27 @@ void AuthenticatorImpl::handleGoodJwt(bool cache_hit) {
     }
   }
 
+  // Copy JWT Claim to Header
+  if (provider.claim_to_header_size() != 0) {
+    for (const auto& headerAndClaim : provider.claim_to_header()) {
+      if (!headerAndClaim.claim_name().empty() && !headerAndClaim.header_name().empty()) {
+        StructUtils payload_getter(jwt_->payload_pb_);
+        std::string stringClaimValue;
+        uint64_t intClaimValue = 0;
+        bool boolClaimValue;
+        if (payload_getter.GetString(headerAndClaim.claim_name(), &stringClaimValue) == StructUtils::OK) {
+          headers_->addCopy(Http::LowerCaseString(headerAndClaim.header_name()), stringClaimValue);
+        } else if(payload_getter.GetUInt64(headerAndClaim.claim_name(), &intClaimValue) == StructUtils::OK) {
+          headers_->addCopy(Http::LowerCaseString(headerAndClaim.header_name()), std::to_string(intClaimValue));
```

Comment:
this is not efficient with 3 lookup.    maybe you can add 
```
// This will fetch the field with primitive type, double, bool and string into a string
GetPrimitiveFieldToString(const string& name, string* out_string);
```
into jwt_verify_lib  StructUtil class